app/src/main/java/org/mozilla/focus/screenshot/ScreenshotViewerActivity.java

```diff
@@ -287,6 +287,12 @@ private void initInfoItemArray() {
     private void initScreenshotInfo(boolean withShare) {
         if (mScreenshot != null) {
             new ScreenshotInfoTask(this, mScreenshot, mInfoItems, withShare).execute();
+            // getCategory may fetch data from internet, isolate it from ScreenshotInfoTask to avoid displaying screenshot lately due to network operation delay
+            ThreadUtils.postToBackgroundThread(() -> {
+                mScreenshot.setCategory(ScreenshotManager.getInstance().getCategory(ScreenshotViewerActivity.this, mScreenshot.getUrl()));
+                mScreenshot.setCategoryVersion(ScreenshotManager.getInstance().getCategoryVersion());
+                mInfoItems.get(4).title = ScreenshotViewerActivity.this.getString(R.string.screenshot_image_viewer_dialog_info_category, mScreenshot.getCategory());
```

Comment:
how do you make sure get(4) will always return what you want?

Reply:
Per discussed, don't update title here. Only query categories.